src/linter.rs

```diff
@@ -91,13 +95,17 @@ impl Linter {
     rules: Arc<Vec<Box<dyn LintRule>>>,
     plugins: Arc<Vec<Box<dyn Plugin>>>,
   ) -> Self {
+    let check_unknown_rules = rules
+      .iter()
+      .any(|a| a.code() == (BanUnknownRuleCode).code());
     Linter {
       ast_parser: AstParser::new(),
       ignore_file_directive,
       ignore_diagnostic_directive,
       syntax,
       rules,
       plugins,
+      check_unknown_rules,
```

Comment:
In my opinion, I don't want the `Linter` struct to have `check_unknown_rules`. Can we store this boolean in `Context` instead?